### ISSUE TITLE
【bug】Error Parser : [ 'permits' cannot be used as an identifier as it is a keyword., 'permits' cannot be used as an identifier as it is a keyword ]

### ISSUE THREAD

**Issue by RTxin** [javaparser/javaparser#4490]
JP Veresion:  3.25.10
[scan project uri](https://github.com/apache/pulsar)
```java
    private void upsertLookupPermits(int permits) throws Exception {
        pulsar.getPulsarResources().getDynamicConfigResources().setDynamicConfigurationWithCreate(optMap -> {
            Map<String, String> map = optMap.orElse(new TreeMap<>());
            map.put("maxConcurrentLookupRequest", Integer.toString(permits));
            return map;
        });
    }
```
In the above code,JP identifies the variable name "permits" as the keyword "permits". 
However, the keyword `permits` must be used to decorate a class with the keyword `sealed ` at the same time, which cannot be used independently.
 I think a more detailed distinction should be made for this kind of keyword.
In Intelij IDEA, it does something extra for this.
```java
public sealed interface Solution permits Main{
}
```

**Comment by jlerbsc** [comment]
Please send us a complete test case so that we can reproduce your problem.

**Comment by RTxin** [comment]
[jpTest.zip](https://github.com/user-attachments/files/16124971/jpTest.zip)

@jlerbsc Hello,this is my demo

**Comment by jlerbsc** [comment]
I am not reproducing your issue with this test case. Can you improve this test to simply reproduce the problem?

    @Test
	void issue4490() {
		String code =
				"class Foo {\n"
				+ "    void m(int permits) {}\n"
				+ "}";
		JavaParser javaParser = new JavaParser();
		javaParser.getParserConfiguration().setLanguageLevel(ParserConfiguration.LanguageLevel.JAVA_18);
		ParseResult<CompilationUnit> result = javaParser.parse(code);

		if (!result.isSuccessful()) {
			for (Problem problem : result.getProblems()) {
				System.out.println(problem.getMessage());
			}
		}
	}

**Comment by RTxin** [comment]
> I am not reproducing your issue with this test case. Can you improve this test to simply reproduce the problem?
> 
> ```
> @Test
> void issue4490() {
> 	String code =
> 			"class Foo {\n"
> 			+ "    void m(int permits) {}\n"
> 			+ "}";
> 	JavaParser javaParser = new JavaParser();
> 	javaParser.getParserConfiguration().setLanguageLevel(ParserConfiguration.LanguageLevel.JAVA_18);
> 	ParseResult<CompilationUnit> result = javaParser.parse(code);
> 
> 	if (!result.isSuccessful()) {
> 		for (Problem problem : result.getProblems()) {
> 			System.out.println(problem.getMessage());
> 		}
> 	}
> }
> ```
I am sorry that I did not provide you with a simple case, but the case you have given can already reproduce this problem

**Comment by jlerbsc** [comment]
> but the case you have given can already reproduce this problem

But that's not the case!

**Comment by RTxin** [comment]
> > but the case you have given can already reproduce this problem
> 
> But that's not the case!
Does the sentence in the picture not appear when you run case code?
![image](https://github.com/javaparser/javaparser/assets/45174036/36f82015-89e7-40f3-b210-1accd2ffaf4f)

**Comment by jlerbsc** [comment]
There is no problem displayed

**Comment by RTxin** [comment]
> There is no problem displayed

I got it, you already fixed it in #4039 in versionversion3.25.4 and I'm using 3.25.1.
Just searched the release logs and realised it's been fixed, sorry for wasting your time!❤🌹

**Comment by RTxin** [comment]
> There is no problem displayed

em...... After testing and searching the release notes, the issue has been fixed in version 3.26.0 #4434. However, version 3.25.10 should be able to reproduce it.
But I'm sorry for wasting your time because of my laziness in not checking the release notes.❤🌹

**Comment by jlerbsc** [comment]
I hadn't noticed which version of JP you were using
